sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIOPatientEverything.java

```diff
@@ -0,0 +1,237 @@
+      new TupleTag<HealthcareIOError<String>>() {};
+
+  /**
+   * PatientEverythingParameter defines required attributes for a FHIR GetPatientEverything request
+   * in {@link FhirIOPatientEverything}. *
+   */
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class PatientEverythingParameter implements Serializable {
+
+    /**
+     * FHIR Patient resource name in the format of
+     * projects/{p}/locations/{l}/datasets/{d}/fhirStores/{f}/fhir/{resourceType}/{id}.
+     */
+    abstract String getResourceName();
+    /** Optional filters for the request, eg. start, end, _type, _since, _count */
+    abstract @Nullable Map<String, String> getFilters();
+
+    static Builder builder() {
+      return new AutoValue_FhirIOPatientEverything_PatientEverythingParameter.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setResourceName(String resourceName);
+
+      abstract Builder setFilters(Map<String, String> query);
+
+      abstract PatientEverythingParameter build();
+    }
+  }
+
+  /** The Result for a {@link FhirIOPatientEverything} request. */
+  public static class Result implements POutput, PInput {
+
+    private final PCollection<JsonArray> patientCompartments;
+    private final PCollection<HealthcareIOError<String>> failedReads;
+
+    PCollectionTuple pct;
+
+    /**
+     * Create FhirIOPatientEverything.Result form PCollectionTuple with OUT and DEAD_LETTER tags.
+     *
+     * @param pct the pct
+     * @return the patient everything result
+     * @throws IllegalArgumentException the illegal argument exception
+     */
+    static Result of(PCollectionTuple pct) throws IllegalArgumentException {
+      if (pct.has(OUT) && pct.has(DEAD_LETTER)) {
+        return new Result(pct);
+      } else {
+        throw new IllegalArgumentException(
+            "The PCollection tuple must have the FhirIOPatientEverything.OUT "
+                + "and FhirIOPatientEverything.DEAD_LETTER tuple tags");
+      }
+    }
+
+    private Result(PCollectionTuple pct) {
+      this.pct = pct;
+      this.patientCompartments = pct.get(OUT).setCoder(JsonArrayCoder.of());
+      this.failedReads =
+          pct.get(DEAD_LETTER).setCoder(HealthcareIOErrorCoder.of(StringUtf8Coder.of()));
+    }
+
+    /**
+     * Gets failed reads.
+     *
+     * @return the failed reads
+     */
+    public PCollection<HealthcareIOError<String>> getFailedReads() {
+      return failedReads;
+    }
+
+    /**
+     * Gets the patient compartment responses for GetPatientEverything requests.
+     *
+     * @return the read patient compartments
+     */
+    public PCollection<JsonArray> getPatientCompartments() {
+      return patientCompartments;
+    }
+
+    @Override
+    public Pipeline getPipeline() {
+      return this.pct.getPipeline();
+    }
+
+    @Override
+    public Map<TupleTag<?>, PValue> expand() {
```

Comment:
are you interested in returning a `Result` type with `getFailedReads` and `getPatientCompartments`? This would be a little nicer to use for others, but maybe you have your reasons for going with the Map?

Reply:
Sorry... isn't this what I'm doing? The return type of the FhirIOPatientEverything PTransform is Result, and there is the methods getFailedReads and getPatientCompartments, but the type POutput has to implement `expand`, like all the other .Result methods in FhirIO?